### ISSUE TITLE
[QUESTION] Kinesis Libs built on MacOS not working with iOS Application

### ISSUE THREAD

**Issue by mehmetcy84** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#1545]
I have built and ran the samples for Kinesis C SDK on macOS 12.5. I would like to use a C based code similar to the sample client code, and use the built Kinesis libraries in my iOS application.

I get the following errors/warnings when I add the built libraries to my iOS XCode project (.dylib and .a files, e.g. libssl.dylib, libsrtp2.a etc.) and try to build for
a) iOS simulator target:
_ld: building for iOS Simulator, but linking in dylib built for macOS, file '/libKinesis/libssl.dylib' for architecture x86_64
b) iPhone 12 device:
ld: warning: ignoring file /libKinesis/libkvspicUtils.a, building for iOS-arm64 but attempting to link with file built for macOS-x86_64

and build fails for both cases.

So my questions are,
1) Is it possible to use the Kinesis C SDK libraries as is for the iOS application (simulator and real device)?
2) Is it possible to use the Kinesis C SDK library with iOS applications? Has this use case been tested and verified? (I can already use third party C repos with my iOS applications.)
3) Do I need to modify some Makefiles to specifically set the target as iOS instead of macOS while building the Kinesis SDK?
4) Am I forced to use the Kinesis iOS SDK specifically? https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-ios/
I would really prefer to be able to use my C based repository, which I can share among different platforms. (Windows, iOS)

Thanks and best regards,

**Comment by hassanctech** [comment]
Yes you can use to C SDK.  You will need to set up your tool chain to compile for iOS, MacOS is a different platform, builds for MacOS will not run on iOS, but Apple provides these toolchains to build your application on iOS.  Since this isn't a KVS specific question I will close this issue but I encourage you to post on Apple/iOS forums to understand how to build a standard C project for iOS, that will help you to build for iOS.  In general since you're doing a cross compile we have an example in our CI:

https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/master/.github/workflows/ci.yml#L327-L344

But I think what would help you more is to visit iOS development forums and just find out how to build any old C library for iOS from your Mac and then replicate the process for this library.  If at that point you run into issues please post back with a new question detailing the issues.
